### PR TITLE
[pt2e] Avoid getting model device once per node

### PR DESCRIPTION
**Pull request by andrewor14**
**Summary:** Previously, we call `assert_and_get_unqiue_device` once per node in both prepare and convert. This is expensive and unnecessary since the model device is the same across all nodes, so we should just call this once in the beginning and reuse the same model device across all the nodes.

**Test Plan:**
python test/test_quantization.py -k TestQuantizePT2E

cc @ezyang @SherlockNoMad @EikanWang @jgong5 @wenzhe-nrv

### CODE REVIEW

File: torch/ao/quantization/utils.py [hunk 1]
```diff
@@ -800,15 +800,6 @@ def _assert_and_get_unique_device(module: torch.nn.Module) -> Any:
...
     }
-    """
-    As a temp workaround for AIMP HHC publish we added CPU check.remove it later. T163614564
-    """
-    if {torch.device("cpu"), torch.device("meta")} == devices:
-        warnings.warn(
-            "Both 'meta' and 'cpu' are present in the list of devices. Module can have one device. We Select 'cpu'."
-        )
-        devices = {torch.device("cpu")}
-    ""
```
Comment (andrewor14) [inline torch/ao/quantization/utils.py]: @jerryzh168 is it safe to remove this?

Comment (jerryzh168) [reply torch/ao/quantization/utils.py]: let me check

**Comment by iremyux** [comment]
Adding `ciflow/win-arm64` label to trigger Windows Arm64 CI and its test purposes - nothing about this PR specifically. (It should not effect the acceptance of the PR even if it fails.)

File: torch/ao/quantization/observer.py [hunk 2]
```diff
@@ -1902,10 +1902,18 @@ def convert(self, model: torch.fx.GraphModule, observer_node: Node):
             else:
                 scale, zero_point = self.calculate_qparams()
                 scale_node = create_getattr_from_value(
-                    model, model.graph, "_scale", scale
+                    model,
+                    model.graph,
+                    "_scale",
+                    scale,
+                    scale.device,
```
Comment (jerryzh168) [inline torch/ao/quantization/observer.py]: value could be a non-Tensor I think? according to `create_getattr_from_value` L274, so should this be the device from model?

Comment (andrewor14) [reply torch/ao/quantization/observer.py]: Ok, seems like if we just get it from the model there will be no benefits, so how about here we get it from the scale if it's a tensor and get it from the model otherwise?

**Review by jerryzh168 — APPROVED** [review]
